app/src/auth/components/SessionProvider/index.jsx

```diff
@@ -0,0 +1,63 @@
+// @flow
+
+import React, { type Node } from 'react'
+
+import Context, { type Props as State } from '$auth/contexts/Session'
+import { isSessionStorageAvailable } from '$mp/utils/storage'
+
+const SESSION_TOKEN_KEY = 'session.token'
+
+type Props = {
+    children: Node,
+}
+
+const storage = isSessionStorageAvailable() ? sessionStorage : null
+
+class SessionProvider extends React.Component<Props, State> {
+    static token(): ?string {
+        if (storage) {
+            return storage.getItem(SESSION_TOKEN_KEY) || null
+        }
+        return null
+    }
+
+    static storeToken(value?: ?string) {
+        if (storage) {
+            if (!value) {
+                storage.removeItem(SESSION_TOKEN_KEY)
+            } else {
+                storage.setItem(SESSION_TOKEN_KEY, value)
+            }
+        }
+    }
+
+    state = {
+        setSessionToken: this.setSessionToken.bind(this),
+        token: this.constructor.token(),
+    }
+
+    setSessionToken(token: ?string) {
+        this.constructor.storeToken(token)
+        this.setState({
+            token,
+        })
+    }
+
+    render() {
+        const { children } = this.props
+        const { token, setSessionToken } = this.state
+
+        return (
+            <Context.Provider
+                value={{
```

Comment:
this will always rerender, why not `this.state`?

Reply:
Ok., I'll go with `this.state` used directly and eslint-disable `react/no-unused-state`. Still better than over-rendering.